### PR TITLE
Integrate with gowoken-kicker

### PR DESCRIPTION
**Pull request by magicalne**
<del>Only `whitelist valid test` works as expected. The others still need some work.<'del>

---

All of the tests can pass on godwoken devnet.
We have two major test cases: whitelist and erc20

### CODE REVIEW

**Comment by RetricSu** [comment]
> Only `whitelist valid test` works as expected. The others still need some work.

since we have update the payload format to include func selector, I think we can remove the fallback contract method and related tests. this can be done in another PR, of course

File: test/gasless_paymaster.test.ts [hunk 1]
```diff
@@ -13,73 +13,85 @@ import {
...
-    await paymaster.addStake(99999999, { value: parseEther('2') })
-    await paymaster.addWhitelistAddress(whitelistUser.address)
-    await entryPoint.depositTo(paymaster.address, { value: parseEther('1') })
   })
-    // DummyContract.test(1, 1)
-    const callData = "0x4dd3b30b00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001"
-  describe('#validatePaymasterUserOp', () => {
+  describe('#Whitelist', () => {
     it('whitelist valid', async () => {
+      await paymaster.addStake(99999999, { value: parseEther('2') })
```
Comment (RetricSu) [inline test/gasless_paymaster.test.ts]: why move such operations here? let it stay in the `before_each`, every test can be reused?

Comment (magicalne) [reply test/gasless_paymaster.test.ts]: fixed

**Comment by magicalne** [comment]
`whitelist valid with plain tx` is still failing...

**Comment by RetricSu** [comment]
> `whitelist valid with plain tx` is still failing...

ci is passed...did you mean failed at kicker environments?

**Comment by magicalne** [comment]
> > `whitelist valid with plain tx` is still failing...
> 
> ci is passed...did you mean failed at kicker environments?

yes ..

**Comment by RetricSu** [comment]
it seems one test is still falied for hardhat network? https://github.com/godwokenrises/account-abstraction/actions/runs/3656169960/jobs/6178302203#step:7:232

**Comment by magicalne** [comment]
> it seems one test is still falied for hardhat network? https://github.com/godwokenrises/account-abstraction/actions/runs/3656169960/jobs/6178302203#step:7:232

That's fine I guess. Our erc20 sudt proxy contract only works on godwoken.

**Review by RetricSu — APPROVED** [review]

**Pull request merged**